src/Symfony/Component/HttpKernel/Controller/ArgumentResolver.php

```diff
@@ -99,7 +105,16 @@ public function getArguments(Request $request, callable $controller, ?\Reflectio
                 }
             }
 
-            throw new \RuntimeException(sprintf('Controller "%s" requires that you provide a value for the "$%s" argument. Either the argument is nullable and no null value has been provided, no default value has been provided or there is a non-optional argument after this one.', $this->getPrettyName($controller), $metadata->getName()));
+            $reasons = [];
+            $reasonCounter = 1;
+            foreach ($valueResolverExceptions as $e) {
+                $reasons[] = $reasonCounter.') '.$e->getMessage();
+                ++$reasonCounter;
+            }
+
+            $reasons[] = ($reasonCounter > 1 ? $reasonCounter.') ' : '').'Either the argument is nullable and no null value has been provided, no default value has been provided or there is a non-optional argument after this one.';
```

Comment:
What about adding this only when $reasons is empty? I don't feel like this message is useful in addition to the other (when there's one)